File: LeetCode/Medium/0036-valid-sudoku/0036-valid-sudoku.js
/**
 * @param {character[][]} board
 * @return {boolean}
 */
var isValidSudoku = function (board) {
  const len = 9;
  // #1. 가로
  for (let i = 0; i < 9; i++) {
    const target = board[i];
    const cnt = new Array(9).fill(false);
    for (const t of target) {
      if (t === ".") continue;
      else {
        if (cnt[parseInt(t) - 1] === true) return false;
        cnt[parseInt(t) - 1] = true;
      }
    }
  }
  // #2. 세로
  for (let col = 0; col < 9; col++) {
    const cnt = new Array(9).fill(false);
    for (let row = 0; row < 9; row++) {
      const target = board[row][col];
      if (target === ".") continue;
      else {
        if (cnt[parseInt(target) - 1] === true) return false;
        cnt[parseInt(target) - 1] = true;
      }
    }
  }
  // #3. 3x3
  for (let row_blk = 0; row_blk < 3; row_blk++) {
    for (let col_blk = 0; col_blk < 3; col_blk++) {
      const cnt = new Array(9).fill(false);
      let i = 0,
        j = 0;
      for (i = 0; i < 3; i++) {
        for (j = 0; j < 3; j++) {
          const target = board[row_blk * 3 + i][col_blk * 3 + j];
          if (target === ".") continue;
          else {
            if (cnt[parseInt(target) - 1] === true) return false;
            cnt[parseInt(target) - 1] = true;
          }
        }
      }
    }
  }

  return true;
};

File: LeetCode/Medium/0036-valid-sudoku/0036-valid-sudoku.py
class Solution(object):
    # 배열이 1~9 하나씩만 가지고 있는지 판단하는 코드
    def isValid(self, array):
        s = set()
        for num in array:
            if(num != "."):
                if num not in s:
                    s.add(num)
                else:
                    return False
        return True


    def isValidSudoku(self, board):
        """
        :type board: List[List[str]]
        :rtype: bool
        """
        LENGTH = 9
        #1. 가로 배열이 Valid한지 판단
        for row in range(LENGTH):
            arr = board[row]
            if self.isValid(arr) == False:
                return False
        #2. 세로 배열이 Valid한지 판단
        for col in range(LENGTH):
            arr = [board[row][col] for row in range(LENGTH)]
            if self.isValid(arr) == False:
                return False
        #3. 3 * 3 배열이 Valid 한지 판단
        i, j = 0, 0
        while(i < 3):
            arr = [board[row][col] for row in range(3*i, 3*(i+1)) for col in range (3*j, 3*(j+1))]
            if self.isValid(arr) == False:
                return False
            j += 1
            if(j==3):
                i += 1
                j = 0
        return True
